base/utf16.jl

```diff
@@ -136,6 +281,9 @@ function convert(T::Type{UTF16String}, bytes::AbstractArray{UInt8})
     UTF16String(d)
 end
 
+convert(::Type{UTF16String}, str::UTF16String)    = str
```

Comment:
Yes, but this was about fixing bugs, not removing stuff, I was told many times not to do too many things in a single PR, so I was planning on addressing that after hopefully these bug fix PRs get in.